cli/src/wallet_cli/mod.rs

```diff
@@ -344,6 +347,32 @@ pub async fn address_command(wallet: &Wallet) -> Result<(), Error> {
     Ok(())
 }
 
+// `allot-mana` command
+pub async fn allot_mana_command(wallet: &Wallet, mana: u64, account_id: Option<AccountId>) -> Result<(), Error> {
+    let wallet_data = wallet.data().await;
+    let account_id = account_id
+        .or(wallet_data
+            .accounts()
+            .next()
+            .map(|o| o.output.as_account().account_id_non_null(&o.output_id)))
+        .or(wallet_data
+            .implicit_accounts()
+            .next()
+            .map(|o| AccountId::from(&o.output_id)))
+        .ok_or(WalletError::AccountNotFound)?;
```

Comment:
no check necessary whether a potentially provided account id is contained in `wallet_data.accounts()` or `walllet_data.implicit_accounts`?

Reply:
Not sure what you mean

Reply:
my question basically is: the `account_id` we pass in to the method can by *any* account_id, right? It doesn't have to be related to `wallet.accounts()` whatsoever? It just strikes me as a bit weird logic, that in case we don't provide anything it makes the decision on its own to select either the first account or the first implicit account. But maybe I am just missing something.

Reply:
It just selects a default one if none is provided, makes it easier

Reply:
sure, I guess I just don't see the full picture yet ... if the default is not surprising to the user then it should be fine, I guess. And one last question for my understanding: so if the user provides an `account_id` it will just allot the mana to it without restrictions, otherwise it will allot it to one of its own accounts?

Reply:
I don't think one of your accounts as default is surprising, I'd even say it's expected.
And yes to your question.